### PR TITLE
Add cf-utils folder.

### PR DESCRIPTION
**Pull request by boaks**
Introduce bom module.
Move cf-nat and cf-unix-health into cf-utils.
Move californium-integration-tests and
californium-interoperability-tests into tests.

Signed-off-by: Achim Kraus <achim.kraus@bosch.io>

### CODE REVIEW

**Comment by boaks** [comment]
This PR seems to require a "parent bootstrap".
I think, it's required to introduce a "bom" module as well.

**Comment by boaks** [comment]
@sophokles73 

It would be great, if you can check if the introduced "bom" module complies to common patterns. I tried to implement it according the bom in hono.

File: cf-utils/cf-nat/pom.xml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>org.eclipse.californium</groupId>
+		<artifactId>cf-bom</artifactId>
+		<version>2.3.0-SNAPSHOT</version>
+		<relativePath>../../bom</relativePath>
+	</parent>
```
Comment (sophokles73) [inline cf-utils/cf-nat/pom.xml:13]: Is there a particular reason why you do not use *cf-utils* as the parent?

Comment (boaks) [reply cf-utils/cf-nat/pom.xml:13]: My intention is to use the folder just as collection. If a pom is added to the "cf-utils" folder, that must be published as well in order to be able to us the cf-nat (see issue #1041 ).  Though the modules may vary much more than others "cf-nat" (independent) and (new to come) "cf-cli" (basic command line interface with picocli, highly dependent on californium), the pom on cf-utils will not contain too much information.

**Pull request merged**